exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -335,6 +351,50 @@ private void javaPropertiesToSystemOptions() {
     }
   }
 
+
+  // Polls for graceful file to check if graceful shutdown is triggered from the script.
+  private static class PollShutdownThread extends Thread {
+
+    private final Drillbit drillbit;
+    private final StackTrace stackTrace;
+
+    public PollShutdownThread(final Drillbit drillbit, final StackTrace stackTrace) {
+      this.drillbit = drillbit;
+      this.stackTrace = stackTrace;
+    }
+
+    @Override
+    public void run () {
+      try {
+        pollShutdown(drillbit);
+      } catch (Exception e) {
+        throw new RuntimeException("Caught exception while polling for shutdown\n" + stackTrace, e);
+      }
```

Comment:
I think you have to catch `InterruptedException` and consume it after logging rather than throwing `RuntimeException` for it. Since when `Drillbit.close` is called from another path it will interrupt this thread and we should not throw `RuntimeException` in that case.

Reply:
Changed it